File: frontend/src/services/authService.js
import axios from 'axios';
import config from '../config/config';

const API_URL = `${config.API_BASE_URL}${config.API_ENDPOINTS.AUTH}`;

const signup = (firstName, lastName, email, password) => {
  return axios.post(`${API_URL}/signup`, {
    firstName,
    lastName,
    email,
    password,
  });
};

const signin = (email, password) => {
  return axios
    .post(`${API_URL}/signin`, {
      email,
      password,
    })
    .then((response) => {
      if (response.data.data.token) {
        localStorage.setItem('user', JSON.stringify(response.data.data));
      }
      return response.data.data;
    });
};

const logout = () => {
  localStorage.removeItem('user');
};

const getCurrentUser = () => {
  return JSON.parse(localStorage.getItem('user'));
};

const authService = {
  signup,
  signin,
  logout,
  getCurrentUser,
};

export default authService;

File: url_shortener/src/main/java/com/url_shortener/service/AuthService.java
package com.url_shortener.service;

import com.url_shortener.dto.*;
import com.url_shortener.entity.User;
import com.url_shortener.response.ApiResponse;
import com.url_shortener.response.AuthResponse;

public interface AuthService {
    ApiResponse<AuthResponse> signUp(SignUpRequest request);
    ApiResponse<AuthResponse> signIn(SignInRequest request);
    ApiResponse<User> getCurrentUser(String email);
}
